guake/guake_app.py

```diff
@@ -1096,6 +1107,88 @@ def set_final_window_rect(self):
+                            '/usr/bin/dconf', 'read',
+                            '/org/gnome/shell/extensions/dash-to-dock/multi-monitor'
+                        ]).decode("utf-8").rstrip() or "true"
+
+                    if multi_monitor == "true":
+                        dock_active = True
+                    else:
+                        preferred_monitor = int(
+                            subprocess.check_output([
+                                '/usr/bin/dconf', 'read',
+                                '/org/gnome/shell/extensions/dash-to-dock/preferred-monitor'
+                            ])
+                        )
+                        #check if we are on preferred monitor
+                        if monitor == preferred_monitor:
+                            dock_active = True
+
+                    ## If active, check if and where the dock is shown ##
+                    dock_shown = False
+
+                    # If active, check if auto-hide is on
+                    if dock_active:
+                        log.debug("Gnome dock active")
+                        #check if auto-hide is on
+                        dock_fixed = subprocess.check_output([
+                            '/usr/bin/dconf', 'read',
+                            '/org/gnome/shell/extensions/dash-to-dock/dock-fixed'
+                        ]).decode("utf-8").rstrip() or "true"
+                        if dock_fixed == "false":
+                            auto_hide = True
+                        else: auto_hide = False
+
+                        #If actually shown
+                        if not auto_hide:
+                            #Check dock's position
+                            dock_pos = subprocess.check_output([
+                                '/usr/bin/dconf', 'read',
+                                '/org/gnome/shell/extensions/dash-to-dock/dock-position'
+                            ]).decode("utf-8").rstrip() or "'LEFT'"
+
+                            log.debug("dock_pos = %s", dock_pos)
+                            if dock_pos == "'LEFT'" or dock_pos == "'RIGHT'":
+                                log.debug("dock_pos is LEFT or RIGHT")
+                                dock_size = int(
+                                    subprocess.check_output([
+                                        '/usr/bin/dconf', 'read',
+                                        '/org/gnome/shell/extensions/dash-to-dock/dash-max-icon-size'
+                                    ])
+                                )
+                                ## Adjust width ##
+                                log.debug("Adjusting window size")
+                                dock_shown = True
+                                dock_size += 19
+                                window_rect.width = window_rect.width - dock_size
+                                if monitor == 1:
+                                    window_rect.x += dock_size                       
```

Comment:
This is what, **nineteen** nesting levels?..

That's too bad; please see if you can refactor the code into something neater.

Reply:
Ok, I'll check it tomorrow

Reply:
I commited the new version with less nesting levels. 
Basically I serialized all the if statements. Don't really like it conceptually, but I guess it's a matter of taste.

Please note that the new code is UNTESTED
(I'm confined at home for the coronavirus, so I'm working on different machine with no double monitor).